cookies/views.py

```diff
@@ -1136,3 +1136,46 @@ def trigger_giles_submission(request, resource_id, relation_id):
                              " connecting to the redis message passing"
                              " backend.")
             return HttpResponse(task)
+
+
+@login_required
+def bulk_action_resource(request):
+    """
+    Curator can perform actions with resources selected
+    """
+    resource_ids = request.POST.getlist('addresources', [])
+    qs = Resource.objects.filter(pk__in=resource_ids)
+
+    qset_collections = Collection.objects.filter(
+        Q(content_resource=False)\
+        & Q(hidden=False) & (Q(public=True) | Q(created_by_id=request.user.id))
+    )
+    collections = CollectionFilter(request.GET, queryset=qset_collections)
+
+    context = RequestContext(request, {
+    'collections': qset_collections,
+    'resources': qs,
+    'number_of_resources': qs.count
+    })
+
+    template = loader.get_template('add_resources_to_collection.html')
+    return HttpResponse(template.render(context))
+
+
+@login_required
+def bulk_add_resource_to_collection(request):
+    """
+    Curator adds resource to collection
+    """
+    resource_ids = request.POST.getlist('addresources', [])
+    if len(resource_ids) < 1:
+        raise ValueError('Need more than one resource')
+
+    qs = Resource.objects.filter(pk__in=resource_ids)
+
+    master_id = request.POST.get('master', None)
```

Comment:
The logic here looks good. At the moment this code is a bit hard to read unless one already knows what's going on -- one thing that will help is choosing more informative variable names. E.g. it's hard to infer from the name ``master`` that we're working with a ``Collection`` instance.